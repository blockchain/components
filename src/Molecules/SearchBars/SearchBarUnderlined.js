// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
import styled from 'styled-components'

import { FasSearch } from '../../Atoms/Icons'
import { Input } from '../../Atoms/Inputs'
import injectE2E from '../../Tools/injectE2E'
import { prop, theme } from '../../Tools/interpolation'

export type PropsType = {
  +children?: React.Node,
  +e2e?: string,
  +onChange?: (SyntheticEvent<HTMLInputElement>) => void,
  +onClick?: (SyntheticEvent<HTMLInputElement>) => void,
  +value?: string | number,
  +width?: string,
}

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: ${prop<PropsType>('width')};
  border-bottom: 1px solid ${theme('searchBarUnderlinedBorderColor')};
  background-color: ${theme('searchBarUnderlinedBackgroundColor')};
`

const SearchInput = styled(Input)`
  & > input {
    background-color: inherit;
    border-color: transparent;

    &:focus,
    &:active {
      box-shadow: none !important;
    }
  }
`

const IconContainer = styled.div.attrs({
  'data-e2e': injectE2E,
})`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 60px;
  height: 40px;
`

const SearchBarUnderlined = ({
  children,
  e2e,
  onChange,
  onClick,
  value,
  width,
  ...rest
}: PropsType) => (
  <Wrapper width={width}>
    <SearchInput
      e2e={`${e2e || ''}.Input`}
      onChange={onChange}
      value={value}
      {...rest}
    />
    <IconContainer e2e={`${e2e || ''}.Button`} onClick={onClick}>
      <FasSearch selectable size="20px" />
    </IconContainer>
  </Wrapper>
)

SearchBarUnderlined.propTypes = {
  children: PropTypes.node,
  e2e: PropTypes.string,
  onChange: PropTypes.func,
  onClick: PropTypes.func,
  value: PropTypes.any,
  width: PropTypes.string,
}

SearchBarUnderlined.defaultProps = {
  e2e: 'defaultSearchBarUnderlined',
  width: '100%',
}

export default SearchBarUnderlined
