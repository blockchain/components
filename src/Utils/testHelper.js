import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
