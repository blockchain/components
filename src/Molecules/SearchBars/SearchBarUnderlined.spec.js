/* eslint no-console: 0 */
import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import SearchBarUnderlined from './SearchBarUnderlined'

describe('SearchBarUnderlined', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<SearchBarUnderlined />)

    expect(tree.find(SearchBarUnderlined)).toMatchSnapshot()
  })

  it('should render correctly with width', () => {
    const tree = mountWithTheme(
      <SearchBarUnderlined
        onChange={jest.fn()}
        value="My search"
        width="200px"
      />,
    )

    expect(tree.find(SearchBarUnderlined)).toMatchSnapshot()
  })

  it('should render correctly with value', () => {
    const tree = mountWithTheme(
      <SearchBarUnderlined onChange={jest.fn()} value="My search" />,
    )

    expect(tree.find(SearchBarUnderlined)).toMatchSnapshot()
  })

  it('should render correctly with e2e', () => {
    const tree = mountWithTheme(<SearchBarUnderlined e2e={null} />)

    expect(tree.find(SearchBarUnderlined)).toMatchSnapshot()
  })

  it('should render correctly with onChange', () => {
    const tree = mountWithTheme(<SearchBarUnderlined onChange={jest.fn()} />)

    expect(tree.find(SearchBarUnderlined)).toMatchSnapshot()
  })

  it('should render correctly with onClick', () => {
    const tree = mountWithTheme(<SearchBarUnderlined onClick={jest.fn()} />)

    expect(tree.find(SearchBarUnderlined)).toMatchSnapshot()
  })
})
