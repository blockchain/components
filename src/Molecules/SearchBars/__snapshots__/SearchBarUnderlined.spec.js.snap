// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBarUnderlined should render correctly 1`] = `
.c4 {
  fill: #545456;
  cursor: pointer;
}

.c4:hover {
  fill: #757679;
}

.c2 {
  display: block;
  height: 2.7rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: all 0.25s ease-out;
  transition: all 0.25s ease-out;
}

.c2:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c2:active:not(:disabled),
.c2:focus:not(:disabled) {
  box-shadow: 0 0 0 1px #10ADE4;
}

.c2::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border-bottom: 1px solid #CCC;
  background-color: rgba(0,0,0,0);
}

.c1 {
  background-color: inherit;
  border-color: transparent;
}

.c1:focus,
.c1:active {
  box-shadow: none !important;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 60px;
  height: 40px;
}

<SearchBarUnderlined
  e2e="defaultSearchBarUnderlined"
  width="100%"
>
  <styled.div
    width="100%"
  >
    <div
      className="c0"
      width="100%"
    >
      <Styled(Input)
        e2e="defaultSearchBarUnderlined.Input"
      >
        <Input
          className="c1"
          disabled={false}
          e2e="defaultSearchBarUnderlined.Input"
          type="text"
          width="100%"
        >
          <styled.input
            className="c1"
            disabled={false}
            e2e="defaultSearchBarUnderlined.Input"
            type="text"
            width="100%"
          >
            <input
              className="c1 c2"
              data-e2e="defaultSearchBarUnderlined.Input"
              disabled={false}
              type="text"
              width="100%"
            />
          </styled.input>
        </Input>
      </Styled(Input)>
      <styled.div
        e2e="defaultSearchBarUnderlined.Button"
      >
        <div
          className="c3"
          data-e2e="defaultSearchBarUnderlined.Button"
        >
          <FasSearch
            selectable={true}
            size="20px"
          >
            <Styled(SvgFasSearch)
              height="20px"
              selectable={1}
              width="20px"
            >
              <SvgFasSearch
                className="c4"
                height="20px"
                selectable={1}
                width="20px"
              >
                <svg
                  className="c4"
                  height="20px"
                  selectable={1}
                  viewBox="0 0 512 512"
                  width="20px"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  />
                </svg>
              </SvgFasSearch>
            </Styled(SvgFasSearch)>
          </FasSearch>
        </div>
      </styled.div>
    </div>
  </styled.div>
</SearchBarUnderlined>
`;

exports[`SearchBarUnderlined should render correctly with e2e 1`] = `
.c4 {
  fill: #545456;
  cursor: pointer;
}

.c4:hover {
  fill: #757679;
}

.c2 {
  display: block;
  height: 2.7rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: all 0.25s ease-out;
  transition: all 0.25s ease-out;
}

.c2:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c2:active:not(:disabled),
.c2:focus:not(:disabled) {
  box-shadow: 0 0 0 1px #10ADE4;
}

.c2::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border-bottom: 1px solid #CCC;
  background-color: rgba(0,0,0,0);
}

.c1 {
  background-color: inherit;
  border-color: transparent;
}

.c1:focus,
.c1:active {
  box-shadow: none !important;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 60px;
  height: 40px;
}

<SearchBarUnderlined
  e2e={null}
  width="100%"
>
  <styled.div
    width="100%"
  >
    <div
      className="c0"
      width="100%"
    >
      <Styled(Input)
        e2e=".Input"
      >
        <Input
          className="c1"
          disabled={false}
          e2e=".Input"
          type="text"
          width="100%"
        >
          <styled.input
            className="c1"
            disabled={false}
            e2e=".Input"
            type="text"
            width="100%"
          >
            <input
              className="c1 c2"
              data-e2e=".Input"
              disabled={false}
              type="text"
              width="100%"
            />
          </styled.input>
        </Input>
      </Styled(Input)>
      <styled.div
        e2e=".Button"
      >
        <div
          className="c3"
          data-e2e=".Button"
        >
          <FasSearch
            selectable={true}
            size="20px"
          >
            <Styled(SvgFasSearch)
              height="20px"
              selectable={1}
              width="20px"
            >
              <SvgFasSearch
                className="c4"
                height="20px"
                selectable={1}
                width="20px"
              >
                <svg
                  className="c4"
                  height="20px"
                  selectable={1}
                  viewBox="0 0 512 512"
                  width="20px"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  />
                </svg>
              </SvgFasSearch>
            </Styled(SvgFasSearch)>
          </FasSearch>
        </div>
      </styled.div>
    </div>
  </styled.div>
</SearchBarUnderlined>
`;

exports[`SearchBarUnderlined should render correctly with onChange 1`] = `
.c4 {
  fill: #545456;
  cursor: pointer;
}

.c4:hover {
  fill: #757679;
}

.c2 {
  display: block;
  height: 2.7rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: all 0.25s ease-out;
  transition: all 0.25s ease-out;
}

.c2:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c2:active:not(:disabled),
.c2:focus:not(:disabled) {
  box-shadow: 0 0 0 1px #10ADE4;
}

.c2::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border-bottom: 1px solid #CCC;
  background-color: rgba(0,0,0,0);
}

.c1 {
  background-color: inherit;
  border-color: transparent;
}

.c1:focus,
.c1:active {
  box-shadow: none !important;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 60px;
  height: 40px;
}

<SearchBarUnderlined
  e2e="defaultSearchBarUnderlined"
  onChange={[MockFunction]}
  width="100%"
>
  <styled.div
    width="100%"
  >
    <div
      className="c0"
      width="100%"
    >
      <Styled(Input)
        e2e="defaultSearchBarUnderlined.Input"
        onChange={[MockFunction]}
      >
        <Input
          className="c1"
          disabled={false}
          e2e="defaultSearchBarUnderlined.Input"
          onChange={[MockFunction]}
          type="text"
          width="100%"
        >
          <styled.input
            className="c1"
            disabled={false}
            e2e="defaultSearchBarUnderlined.Input"
            onChange={[MockFunction]}
            type="text"
            width="100%"
          >
            <input
              className="c1 c2"
              data-e2e="defaultSearchBarUnderlined.Input"
              disabled={false}
              onChange={[MockFunction]}
              type="text"
              width="100%"
            />
          </styled.input>
        </Input>
      </Styled(Input)>
      <styled.div
        e2e="defaultSearchBarUnderlined.Button"
      >
        <div
          className="c3"
          data-e2e="defaultSearchBarUnderlined.Button"
        >
          <FasSearch
            selectable={true}
            size="20px"
          >
            <Styled(SvgFasSearch)
              height="20px"
              selectable={1}
              width="20px"
            >
              <SvgFasSearch
                className="c4"
                height="20px"
                selectable={1}
                width="20px"
              >
                <svg
                  className="c4"
                  height="20px"
                  selectable={1}
                  viewBox="0 0 512 512"
                  width="20px"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  />
                </svg>
              </SvgFasSearch>
            </Styled(SvgFasSearch)>
          </FasSearch>
        </div>
      </styled.div>
    </div>
  </styled.div>
</SearchBarUnderlined>
`;

exports[`SearchBarUnderlined should render correctly with onClick 1`] = `
.c4 {
  fill: #545456;
  cursor: pointer;
}

.c4:hover {
  fill: #757679;
}

.c2 {
  display: block;
  height: 2.7rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: all 0.25s ease-out;
  transition: all 0.25s ease-out;
}

.c2:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c2:active:not(:disabled),
.c2:focus:not(:disabled) {
  box-shadow: 0 0 0 1px #10ADE4;
}

.c2::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border-bottom: 1px solid #CCC;
  background-color: rgba(0,0,0,0);
}

.c1 {
  background-color: inherit;
  border-color: transparent;
}

.c1:focus,
.c1:active {
  box-shadow: none !important;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 60px;
  height: 40px;
}

<SearchBarUnderlined
  e2e="defaultSearchBarUnderlined"
  onClick={[MockFunction]}
  width="100%"
>
  <styled.div
    width="100%"
  >
    <div
      className="c0"
      width="100%"
    >
      <Styled(Input)
        e2e="defaultSearchBarUnderlined.Input"
      >
        <Input
          className="c1"
          disabled={false}
          e2e="defaultSearchBarUnderlined.Input"
          type="text"
          width="100%"
        >
          <styled.input
            className="c1"
            disabled={false}
            e2e="defaultSearchBarUnderlined.Input"
            type="text"
            width="100%"
          >
            <input
              className="c1 c2"
              data-e2e="defaultSearchBarUnderlined.Input"
              disabled={false}
              type="text"
              width="100%"
            />
          </styled.input>
        </Input>
      </Styled(Input)>
      <styled.div
        e2e="defaultSearchBarUnderlined.Button"
        onClick={[MockFunction]}
      >
        <div
          className="c3"
          data-e2e="defaultSearchBarUnderlined.Button"
          onClick={[MockFunction]}
        >
          <FasSearch
            selectable={true}
            size="20px"
          >
            <Styled(SvgFasSearch)
              height="20px"
              selectable={1}
              width="20px"
            >
              <SvgFasSearch
                className="c4"
                height="20px"
                selectable={1}
                width="20px"
              >
                <svg
                  className="c4"
                  height="20px"
                  selectable={1}
                  viewBox="0 0 512 512"
                  width="20px"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  />
                </svg>
              </SvgFasSearch>
            </Styled(SvgFasSearch)>
          </FasSearch>
        </div>
      </styled.div>
    </div>
  </styled.div>
</SearchBarUnderlined>
`;

exports[`SearchBarUnderlined should render correctly with value 1`] = `
.c4 {
  fill: #545456;
  cursor: pointer;
}

.c4:hover {
  fill: #757679;
}

.c2 {
  display: block;
  height: 2.7rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: all 0.25s ease-out;
  transition: all 0.25s ease-out;
}

.c2:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c2:active:not(:disabled),
.c2:focus:not(:disabled) {
  box-shadow: 0 0 0 1px #10ADE4;
}

.c2::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border-bottom: 1px solid #CCC;
  background-color: rgba(0,0,0,0);
}

.c1 {
  background-color: inherit;
  border-color: transparent;
}

.c1:focus,
.c1:active {
  box-shadow: none !important;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 60px;
  height: 40px;
}

<SearchBarUnderlined
  e2e="defaultSearchBarUnderlined"
  onChange={[MockFunction]}
  value="My search"
  width="100%"
>
  <styled.div
    width="100%"
  >
    <div
      className="c0"
      width="100%"
    >
      <Styled(Input)
        e2e="defaultSearchBarUnderlined.Input"
        onChange={[MockFunction]}
        value="My search"
      >
        <Input
          className="c1"
          disabled={false}
          e2e="defaultSearchBarUnderlined.Input"
          onChange={[MockFunction]}
          type="text"
          value="My search"
          width="100%"
        >
          <styled.input
            className="c1"
            disabled={false}
            e2e="defaultSearchBarUnderlined.Input"
            onChange={[MockFunction]}
            type="text"
            value="My search"
            width="100%"
          >
            <input
              className="c1 c2"
              data-e2e="defaultSearchBarUnderlined.Input"
              disabled={false}
              onChange={[MockFunction]}
              type="text"
              value="My search"
              width="100%"
            />
          </styled.input>
        </Input>
      </Styled(Input)>
      <styled.div
        e2e="defaultSearchBarUnderlined.Button"
      >
        <div
          className="c3"
          data-e2e="defaultSearchBarUnderlined.Button"
        >
          <FasSearch
            selectable={true}
            size="20px"
          >
            <Styled(SvgFasSearch)
              height="20px"
              selectable={1}
              width="20px"
            >
              <SvgFasSearch
                className="c4"
                height="20px"
                selectable={1}
                width="20px"
              >
                <svg
                  className="c4"
                  height="20px"
                  selectable={1}
                  viewBox="0 0 512 512"
                  width="20px"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  />
                </svg>
              </SvgFasSearch>
            </Styled(SvgFasSearch)>
          </FasSearch>
        </div>
      </styled.div>
    </div>
  </styled.div>
</SearchBarUnderlined>
`;

exports[`SearchBarUnderlined should render correctly with width 1`] = `
.c4 {
  fill: #545456;
  cursor: pointer;
}

.c4:hover {
  fill: #757679;
}

.c2 {
  display: block;
  height: 2.7rem;
  width: 100%;
  padding: 0 1rem;
  box-sizing: border-box;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  color: #545456;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.25rem;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  -webkit-transition: all 0.25s ease-out;
  transition: all 0.25s ease-out;
}

.c2:disabled {
  color: #545456;
  border-color: #CCC;
  background: #F8F8F8;
  cursor: not-allowed;
}

.c2:active:not(:disabled),
.c2:focus:not(:disabled) {
  box-shadow: 0 0 0 1px #10ADE4;
}

.c2::-webkit-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::-moz-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2:-ms-input-placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c2::placeholder {
  color: #999B9E;
  font-size: 0.875rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 200px;
  border-bottom: 1px solid #CCC;
  background-color: rgba(0,0,0,0);
}

.c1 {
  background-color: inherit;
  border-color: transparent;
}

.c1:focus,
.c1:active {
  box-shadow: none !important;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 60px;
  height: 40px;
}

<SearchBarUnderlined
  e2e="defaultSearchBarUnderlined"
  onChange={[MockFunction]}
  value="My search"
  width="200px"
>
  <styled.div
    width="200px"
  >
    <div
      className="c0"
      width="200px"
    >
      <Styled(Input)
        e2e="defaultSearchBarUnderlined.Input"
        onChange={[MockFunction]}
        value="My search"
      >
        <Input
          className="c1"
          disabled={false}
          e2e="defaultSearchBarUnderlined.Input"
          onChange={[MockFunction]}
          type="text"
          value="My search"
          width="100%"
        >
          <styled.input
            className="c1"
            disabled={false}
            e2e="defaultSearchBarUnderlined.Input"
            onChange={[MockFunction]}
            type="text"
            value="My search"
            width="100%"
          >
            <input
              className="c1 c2"
              data-e2e="defaultSearchBarUnderlined.Input"
              disabled={false}
              onChange={[MockFunction]}
              type="text"
              value="My search"
              width="100%"
            />
          </styled.input>
        </Input>
      </Styled(Input)>
      <styled.div
        e2e="defaultSearchBarUnderlined.Button"
      >
        <div
          className="c3"
          data-e2e="defaultSearchBarUnderlined.Button"
        >
          <FasSearch
            selectable={true}
            size="20px"
          >
            <Styled(SvgFasSearch)
              height="20px"
              selectable={1}
              width="20px"
            >
              <SvgFasSearch
                className="c4"
                height="20px"
                selectable={1}
                width="20px"
              >
                <svg
                  className="c4"
                  height="20px"
                  selectable={1}
                  viewBox="0 0 512 512"
                  width="20px"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  />
                </svg>
              </SvgFasSearch>
            </Styled(SvgFasSearch)>
          </FasSearch>
        </div>
      </styled.div>
    </div>
  </styled.div>
</SearchBarUnderlined>
`;
